### ISSUE TITLE
Specifying dependencies while installing a specific yang.

### ISSUE THREAD

**Issue by paddyvgc** [sysrepo/sysrepo#2105]
I have two yang models "test1.yang" and "test2.yang", the second model test2.yang has one leafref that points to a leaf in the first model test1.yang.
So test2.yang is dependent on test1.yang and if any configuration delete is happening on the specific node then delete should happen first on test2.yang and then test1.yang (Referred node needs to be deleted first then the referenced node).
If a creation is happening, then the test1.yang needs to be imported first then test2.yang.
Basically my question is where can I specify these dependencies? Is it automatically handled when a leafref is defined  or do I need to specify the dependency while installing. If yes, Please guide me how?

Thanks,

**Comment by michalvasko** [comment]
Hi,
it is performed automatically, *sysrepo* tracks these dependencies on its own.

**Comment by paddyvgc** [comment]
Hi Michal,
Thanks for the response.

How sysrepo will identify the dependency? Is it only for the leafref ones?

So If test2.yang has dependency on test1.yang but no leafref used , so how to specify that the test2.yang is a submodule or a dependent module of test1.yang?
Do I need specify that in yang file?
Is there any significance for sysrepo-module-dependencies.xml here?

**Comment by michalvasko** [comment]
So, you are obviously talking about the legacy *sysrepo*, that is important to mention (it is not really supported anymore). Nevertheless, I still do not understand what your question is. What dependency does `test2` have on `test1` if it s not a `leafref`? There are many other kinds of dependencies and all of them are handled automatically. It does not make sense to enforce some "virtual" dependency, leave it all up to YANG.
